### PR TITLE
Bump wagmi, viem and rainbowkit versions

### PR DESCRIPTION
**Pull request by technophile-04**
## Description

Updated wagmi, viem, rainbowkit to their latest version.  

Wagmi and Viem have pushed some nice experimental freatures for playing around new EIP's in AA world 5792, 3074, 6492 checkout : 
- https://viem.sh/experimental
- https://wagmi.sh/react/api/hooks/useCallsStatus

And also in latest wagmi version, they updated coinbase connector so that coinbase smart wallets can be used checkout: https://x.com/wevm_dev/status/1791218863545126978

### CODE REVIEW

File: packages/nextjs/utils/scaffold-eth/getParsedError.ts
Comment (technophile-04) [inline packages/nextjs/utils/scaffold-eth/getParsedError.ts:1]: There was some update to how viem handles error in latest version, So had to update this file 

Here is demo video of testing : 

https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/64f11c32-39df-43d8-b76c-3774ff1d4d1e

To test locally : 

<details>

<summary>
Copy this in `YourContract.sol` : 
</summary>

```sol
//SPDX-License-Identifier: MIT
pragma solidity >=0.8.0 <0.9.0;

// Useful for debugging. Remove when deploying to a live network.
import "hardhat/console.sol";

error NOT_OWNER();

error MY_CUSTOM_ERROR(string message, uint value);

contract YourContract {
	// State Variables
	address public immutable owner;
	string public greeting = "Building Unstoppable Apps!!!";
	uint256 public totalCounter = 0;

	constructor(address _owner) {
		owner = _owner;
	}

	modifier isOwner() {
		if (msg.sender != owner) {
			revert NOT_OWNER();
		}
		_;
	}

	function revertWithCustomErrror(bool _shouldRevert) public view {
		if (_shouldRevert) {
			revert MY_CUSTOM_ERROR(
				"Error with my custom message",
				totalCounter
			);
		}
	}

	function revertWithRequire(bool _shouldRevert) public pure {
		require(!_shouldRevert, "Error with require message");
	}

	function setGreeting(string memory _newGreeting) public payable {
		// Change state variables
		greeting = _newGreeting;
		totalCounter += 1;
	}

	function withdraw() public isOwner {
		(bool success, ) = owner.call{ value: address(this).balance }("");
		require(success, "Failed to send Ether");
	}

	receive() external payable {}
}

```

</details>

<details>

<summary>
copy this in `page.tsx` to see if other errors are parsed nicely : 
</summary>

```ts
"use client";

import Link from "next/link";
import type { NextPage } from "next";
import { useAccount, useBalance, useSignMessage } from "wagmi";
import { BugAntIcon, MagnifyingGlassIcon } from "@heroicons/react/24/outline";
import { Address } from "~~/components/scaffold-eth";
import { getParsedError, notification } from "~~/utils/scaffold-eth";

const Home: NextPage = () => {
  const { address: connectedAddress } = useAccount();

  const { signMessageAsync } = useSignMessage();

  const { data, error } = useBalance({
    address: connectedAddress,
    chainId: 31337,
  });
  console.log({ data, error });
  console.log("The parsedErrror", getParsedError(error));

  return (
    <>
      <div className="flex items-center flex-col flex-grow pt-10">
        <button
          className="btn btn-primary"
          onClick={async () => {
            try {
              await signMessageAsync({ message: "Hello world !" });
            } catch (e) {
              const erro = getParsedError(e);
              notification.error(erro);
            }
          }}
        >
          Sign Message
        </button>
        <div className="px-5">
          <h1 className="text-center">
            <span className="block text-2xl mb-2">Welcome to</span>
            <span className="block text-4xl font-bold">Scaffold-ETH 2</span>
          </h1>
          <div className="flex justify-center items-center space-x-2">
            <p className="my-2 font-medium">Connected Address:</p>
            <Address address={connectedAddress} />
          </div>
          <p className="text-center text-lg">
            Get started by editing{" "}
            <code className="italic bg-base-300 text-base font-bold max-w-full break-words break-all inline-block">
              packages/nextjs/app/page.tsx
            </code>
          </p>
          <p className="text-center text-lg">
            Edit your smart contract{" "}
            <code className="italic bg-base-300 text-base font-bold max-w-full break-words break-all inline-block">
              YourContract.sol
            </code>{" "}
            in{" "}
            <code className="italic bg-base-300 text-base font-bold max-w-full break-words break-all inline-block">
              packages/hardhat/contracts
            </code>
          </p>
        </div>

        <div className="flex-grow bg-base-300 w-full mt-16 px-8 py-12">
          <div className="flex justify-center items-center gap-12 flex-col sm:flex-row">
            <div className="flex flex-col bg-base-100 px-10 py-10 text-center items-center max-w-xs rounded-3xl">
              <BugAntIcon className="h-8 w-8 fill-secondary" />
              <p>
                Tinker with your smart contract using the{" "}
                <Link href="/debug" passHref className="link">
                  Debug Contracts
                </Link>{" "}
                tab.
              </p>
            </div>
            <div className="flex flex-col bg-base-100 px-10 py-10 text-center items-center max-w-xs rounded-3xl">
              <MagnifyingGlassIcon className="h-8 w-8 fill-secondary" />
              <p>
                Explore your local transactions with the{" "}
                <Link href="/blockexplorer" passHref className="link">
                  Block Explorer
                </Link>{" "}
                tab.
              </p>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default Home;

```

</details>

**Comment by technophile-04** [comment]
Also added `metadataBase` to `getMeatdata` function because we get this warning on `main` branch when you do `yarn start`

Screenshot 2024-05-20 at 11 36 50 AM


Although Vercel says they set default value for `metadataBase`  [here](https://arc.net/l/quote/resdwbxh) but it seems to still give the warning. 

Related discussion : 
- https://github.com/scaffold-eth/scaffold-eth-2/pull/811#issuecomment-2088188469
- https://github.com/vercel/next.js/discussions/57251

**Review by rin-st — APPROVED** [review]
Lgtm! Thanks!

**Review by Pabl0cks — APPROVED** [review]
Looking good to me, TYSM @technophile-04 🔥🙌

===========================

Just realized we don't have a default favicon anymore, but created [an issue](https://github.com/scaffold-eth/scaffold-eth-2/issues/850) to not pollute this PR.

**Comment by technophile-04** [comment]
Tysm all merging this 🙌

**Pull request merged**